src/pages/auth/Register/Register.js

```diff
@@ -0,0 +1,171 @@
+// Register.js
+
+import React, { useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { Link } from 'react-router-dom';
+import * as Yup from 'yup';
+import { AiFillEye, AiFillEyeInvisible } from 'react-icons/ai';
+import { registerThunk } from '../../../slices/authSlice/authSlice';
+import { setNotice, setAlert } from '../../../slices/appSlice/appSlice';
+import ErrorMessage from '../../../components/messages/ErrorMessage';
+
+const RegisterSchema = Yup.object().shape({
+  name: Yup.string().min(2, 'Name is Too Short!').max(50, 'Name is Too Long!').required('Name is Required'),
+  email: Yup.string().email('Invalid email').required('Email is Required'),
+  password: Yup.string().min(6, 'Password is Too Short!').max(50, 'Password is Too Long!').required('Password is Required'),
+  confirmPassword: Yup.string().oneOf([Yup.ref('password'), null], 'Passwords must match'),
+});
+const Register = () => {
+  const dispatch = useDispatch();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [name, setName] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+  const [errors, setErrors] = useState({});
+  const passwordRef = useRef(null);
+  const confirmPasswordRef = useRef(null);
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    RegisterSchema.validate({
+      email, password, name, confirmPassword,
+    }, { abortEarly: false })
+      .then((res) => {
+        dispatch(registerThunk(res)).then((res) => {
+          if (res.error) {
+            dispatch(setAlert(res.payload));
+          } else {
+            dispatch(setNotice('Register successful'));
+          }
+        });
+      }).catch((err) => {
+        const newErrors = {};
+        err.inner.forEach((error) => {
+          newErrors[error.path] = error.message;
+        });
+        setErrors(newErrors);
+      });
+  };
+
+  const showHandler = (e) => {
+    e.preventDefault();
+    setShowPassword(!showPassword);
+    if (passwordRef.current.type === 'password') {
+      passwordRef.current.type = 'text';
+    } else {
+      passwordRef.current.type = 'password';
+    }
+  };
+
+  const showConfirmHandler = (e) => {
+    e.preventDefault();
+    setShowConfirmPassword(!showConfirmPassword);
+    if (confirmPasswordRef.current.type === 'password') {
+      confirmPasswordRef.current.type = 'text';
+    } else {
+      confirmPasswordRef.current.type = 'password';
+    }
+  };
+  return (
+    <section className="bg-gray-50 dark:bg-gray-900">
```

Comment:
- [ ] Please make sure to fix the height of this section to shows up correctelly on mobile screens

![mobile-bg2](https://user-images.githubusercontent.com/95412943/230320007-79fac7bb-ae50-4186-88d0-1e9e3dc2565d.png)